tests/compile-fail/invalid_char.rs

```diff
@@ -0,0 +1,8 @@
+fn main() {
+    assert!(std::char::from_u32(-1_i32 as u32).is_none());
+    let _ = match unsafe { std::mem::transmute::<i32, char>(-1) } { //~ ERROR encountered 4294967295, but expected something in the range 0..=1114111
```

Comment:
char has a range? what's going on here?

Reply:
Yes it does? It must be a valid unicode codepoint.

Reply:
I thought char's value ranges had gaps and were generally totally insane. `1114111` seems a little low, too

Reply:
It has gaps. Remember we do first layout-based validation and then type-based validation. This value is so out of range it is already caught by the first.

But if that's too low, our layout code would be wrong and that would be BAD (TM).

Reply:
The maximal value currently seems to be 0x10ffff. In decimal that's 1114111.

Maybe we should make validation print the range in hexadecimal...